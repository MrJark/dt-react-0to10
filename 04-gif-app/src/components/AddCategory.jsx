// import React from 'react'; // no hace falta pq ya está por default
import { useState } from 'react';

// Tarea: renderizar lo que se manda en el input y llevarlo al setCategories (no conseguido porque no me he acordado de que existian las props)
// Las props pueden llamar los elementos del padre, GifApp, al componente hijo, AddCategory, por tanto, desestruturas las props como category y setCategory
export const AddCategory = ({ onNewCategory }) => {

    
    const [inputValue, setInputValue] = useState();

    const onChangeConst = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault(); // para que no refresh el navegador
        // console.log(inputValue);
        if( inputValue.trim().length <= 1 ) return; // para decir que no añada nada que tenga menos de un caracter, quitando espacios

        // setCategories( categories => [inputValue, ...categories] );// Tarea
        setInputValue(''); // para borrar el input cada vez que das enter
        onNewCategory(inputValue.trim());
    }

    return (
        <form onSubmit={onSubmit}>
            <input 
                type="text" 
                placeholder="Find gifs"
                value={inputValue}
                onChange={ onChangeConst }
            />
        </form>
    )
}
