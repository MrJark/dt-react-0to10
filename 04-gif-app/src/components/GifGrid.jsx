// import React from 'react';
import { getGifs } from './helppers/getGifs'

export const GifGrid = ({ category }) => {

  // No es un abuena pracrica crear esta función ⬇️ aquí dentro, la buena práctica dice que debería estar fuera de este componente para que no se renderice constantemente
  // const getGifs = async( category ) => {

  //   const url = `https://api.giphy.com/v1/gifs/search?api_key=TTP5EKm7haUqI0bIeBC3VuXEBZvb1mNh&q=${category}&limit=20`;
  //   const res = await fetch(url);
  //   const { data } = await res.json();
  //   // console.log(data);

  //   const gifs = data.map( img => ({
  //     id: img.id,
  //     title: img.title,
  //     url: img.images.downsized_medium.url,

  //   }));

  //   console.log(gifs);
  //   return gifs;
  // };

  
  getGifs(category); // NO se DEBE colocar la ejecución de una función directamente dentro de un functional component ya que cada vez que la función se llame, se ejecutará una y otra vez
  

  return (
    <>
        <h3>{category}</h3>
        <p>Hello</p>        

    </>
  )
}
