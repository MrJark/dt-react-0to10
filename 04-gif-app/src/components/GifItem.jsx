

export const GifItem = ({ title, url }) => { // lo que necesitas para mostrar un gif es el titulo y el url 

    console.log({title, url});

  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}
