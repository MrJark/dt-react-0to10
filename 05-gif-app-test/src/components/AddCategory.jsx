import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

    
    const [inputValue, setInputValue] = useState(''); 

    const onChangeConst = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault(); 

        if( inputValue.trim().length <= 1 ) return; 

        
        setInputValue('');
        onNewCategory(inputValue.trim());
    }

    return (
        <form onSubmit={onSubmit}>
            <input 
                type="text" 
                placeholder="Find gifs"
                value={inputValue}
                onChange={ onChangeConst }
            />
        </form>
    )
}
