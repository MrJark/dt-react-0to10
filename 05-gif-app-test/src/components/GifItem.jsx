import PropTypes from 'prop-types'; // 1. Añadir los proptypes

export const GifItem = ({ title, url }) => { 

  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

/*
Tarea: 
1. Añadir el PropTypes (no conseguida)
  a. titulo obligatorio
  b. rul obligatorio
2. Evaluar el snapshot
*/

// 2. hacer obligatorios los titulos u las url
GifItem.PropTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}