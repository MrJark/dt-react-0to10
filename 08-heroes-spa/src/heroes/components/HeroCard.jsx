import { Link } from 'react-router-dom';

const CharactersByHero = ( { alter_ego, characters } ) => { 
    
    // if (alter_ego === characters) return (<></>); // 3ª forma de crear el mismo componente para no repetir los nombres y aquí, solo lo estamos creando en este archivo por tanto, no lo vas a exportar a otos elementos
    // return <p> { characters } </p>;

    return (alter_ego === characters) // 4ª forma de hacer que no se repitan los componentes
        ? <></>
        : <p> { characters } </p>;
}


// las props que necesita la carta son los elementos que las identifican
export const HeroCard = ( { 
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters,
    }) => {

        const heroImg = `/assets/heroes/${id}.jpg`;

        // const charactersByHero = (<p> { characters } </p>); // 2º forma de crear el componente para que no se repitan los nombres

    return (
        <>
            <div className="col">
                <div className="card">

                    <div className="row no-gutters">

                        <div className="col-4">
                            <img src={heroImg} className="card-img" alt={superhero} />
                        </div>

                        <div className="col-8">
                            <div className="card-body">

                                <h5 className="card-title"> { superhero } </h5>

                                <p className="card-text"> { alter_ego } </p>

                                {/* {
                                    (alter_ego !== characters ) &&  <p> { characters } </p> // 1ª forma para que no se repitan los nombres de los personajes si han sido los mismos
                                } */}

                                <CharactersByHero characters={characters} alter_ego={alter_ego}/>

                                <p className="card-text">
                                    <small className="text-muted">{ first_appearance }</small>
                                </p>

                                <Link to={`/hero/${ id }`}>
                                    More info...
                                </Link>
                                
                            </div>
                        </div>

                    </div>

                </div>
            </div>
        </>
    )
}
