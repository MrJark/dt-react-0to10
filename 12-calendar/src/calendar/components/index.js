export * from './Navbar';
export * from './CalendarBoxEvent';