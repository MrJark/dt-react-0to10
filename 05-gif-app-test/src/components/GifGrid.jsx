import { useEffect, useState } from 'react';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {


  const { images, isLoading } = useFetchGifs( category );
  

  return (
    <>
        <h3>{category}</h3>

        {
          isLoading && (<h4>Loading...</h4>) 
        }
        <div className="card-grid">
          {
            
            images.map( ( image ) => (
              <GifItem 
                key={image.id}
                {  ...image }
              />
            ))
          }

          </div>     

    </>
  )
};

// proptipes por la category, de la que depende el componente
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}