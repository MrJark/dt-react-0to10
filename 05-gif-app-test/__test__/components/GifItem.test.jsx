import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('test al componente <GifItem/>', () => {

    const title = 'Dragon Ball';
    const url = 'https://url-cualquiera.com/';

    test('match con el snapshot', () =>{

        const { container } = render( <GifItem title={title} url={url}/>);
        
        expect( container ).toMatchSnapshot();

    });

    test('debería retornar el gif en imagen con el título y rul idicados', () => {

        render( <GifItem title={title} url={url}/>)
        // screen.debug();
        // Forma lenta ya que si quieres evaluar muchos componentes de la img, va a ser eterno. Para ello se desesrtuctura la propia img en los elementos que quieres tener
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        /* */
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('moestrar el título como propiedad vista', () => {

        const { container } = render( <GifItem title={title} url={url}/>);
        expect( screen.getByText(title)).toBeTruthy();

    });

});