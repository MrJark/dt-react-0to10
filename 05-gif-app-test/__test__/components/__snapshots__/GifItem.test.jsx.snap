// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test al componente <GifItem/> match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://url-cualquiera.com/"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
